### PR TITLE
fix: improve CLI number validation and add regression tests

### PR DESCRIPTION
**Pull request by phoekerson**
Ensure numeric inputs are strictly validated against DECIMAL_NUMBER_REGEXP before conversion. Added tests to reject Infinity and hex values in watch options to prevent ambiguous parsing.

**Summary**

This PR fixes CLI number parsing by strictly validating numeric string inputs with DECIMAL_NUMBER_REGEXP before converting them with Number(...).
It also adds regression tests to ensure ambiguous numeric formats like Infinity and hex values (e.g. 0x10) are rejected for watch options.

**What kind of change does this PR introduce?**

Bug fix (CLI argument validation improvement) + test coverage improvements.

**Did you add tests for your changes?**

Yes.
Added/updated tests in test/Cli.basictest.js:

valid decimal/scientific numeric string cases
invalid numeric edge cases
explicit rejection of Infinity and hex-style values in watch options

**Does this PR introduce a breaking change?**

No.
This change only tightens validation for invalid/ambiguous numeric CLI inputs and does not change valid input behavior.

**If relevant, what needs to be documented once your changes are merged or what have you already documented?**
No additional documentation is required.
CLI behavior for valid numeric inputs remains the same; invalid ambiguous formats are now consistently rejected.

**Use of AI**

### CODE REVIEW

File: lib/cli.js [hunk 1]
```diff
@@ -603,6 +603,8 @@ const getExpectedValue = (argConfig) => {
 
 /** @typedef {null | string | number | boolean | RegExp | EnumValue | []} ParsedValue */
 
+const DECIMAL_NUMBER_REGEXP = /^[+-]?(?:\d+\.?\d*|\.\d+)(?:e[+-]?\d+)?$/i;
```
Comment (alexander-akait) [inline lib/cli.js:606]: I see a lot of untested chages around `e` and around `(?:\d+\.?\d*|\.\d+)` group

Comment (phoekerson) [reply lib/cli.js:606]: Added tests for decimal forms (leading dot, trailing dot, explicit +) and scientific notation (e / E, optional +/- on exponent). Added negative cases for incomplete/invalid exponent and malformed decimals so the regexp behavior is fully covered.
